cmd/repo-updater/repos/store.go

```diff
@@ -321,12 +380,191 @@ SET
 RETURNING *
 `
 
+// InsertRepos inserts the given repos and their associated sources.
+// It sets the ID field of each given repo to the value of its corresponding row.
+func (s DBStore) InsertRepos(ctx context.Context, repos ...*Repo) error {
+	records := make([]*repoRecord, 0, len(repos))
+
+	for _, r := range repos {
+		repoRec, err := newRepoRecord(r)
+		if err != nil {
+			return err
+		}
+
+		records = append(records, repoRec)
+	}
+
+	encodedRepos, err := json.Marshal(records)
+	if err != nil {
+		return err
+	}
+
+	q := sqlf.Sprintf(insertReposQuery, string(encodedRepos))
+
+	rows, err := s.db.QueryContext(ctx, q.Query(sqlf.PostgresBindVar), q.Args()...)
+	if err != nil {
+		return errors.Wrap(err, "insert")
+	}
+	defer rows.Close()
+	if err := rows.Err(); err != nil {
+		return err
+	}
+
+	var i int
+	for rows.Next() {
+		if err := rows.Scan(&repos[i].ID); err != nil {
+			return err
+		}
+		i++
```

Comment:
nit: IIRC you can do this
```suggestion
	for ;rows.Next(); i++ {
		if err := rows.Scan(&repos[i].ID); err != nil {
			return err
		}
```